### DISCUSSION TITLE
Set Package Mutability?

### DISCUSSION THREAD

**Discussion by TolinSimpson** [openupm/openupm#3469] [Q&A]
I've noticed that packages installed via openUPM are immutable so I've been creating mutable .unitypackages as a separate release so users can edit package contents without cloning the repo. Why are packages forced to be immutable? From what I was able to scrape from forums it has something to do with how the package manager caches packages based on install origin. Is there a way to prevent this?

**Comment by favoyang** [comment]
Packages installed via register are intended to be immutable. Otherwise, you can not tell if it is changed easily.

Two recommended ways to modify a package,
- Copy the package from the Library/PackageCache/PackageName to Packages/PackageName. It is called "eject a package". Packages under the Packages folder are always mutable.
- Fork and Install from a Git URL, which let you track/merge the changes from upstream easily.
